lib/src/charts/widgets/bar_chart.dart

```diff
@@ -59,81 +55,51 @@ class BarChart extends StatefulWidget {
-    var calculatedMaxYValue = 0.0;
-    // We need to find the min & max y value across every bar series
-    // In case, the user hasn't provided these values, we need to calculate them
-    for (final group in widget.data.barData) {
-      var yValue = group.yValues();
-
-      var minV = double.infinity;
-      var maxV = 0.0;
-
-      if (group is MultiBar && group.arrangement == BarGroupArrangement.stack) {
-        minV = min(minV, 0);
-        // For a stack, the y value of the bar is the total of all bars
-        maxV = max(maxV, yValue.fold(0, (a, b) => a + b.yValue));
-      } else {
-        for (final data in yValue) {
-          minV = min(minV, data.yValue.toDouble());
-          maxV = max(maxV, data.yValue.toDouble());
-        }
-      }
-
-      calculatedMinYValue = min(calculatedMinYValue, minV);
-      calculatedMaxYValue = max(calculatedMaxYValue, maxV);
-      _maxBarsInGroup = max(_maxBarsInGroup, yValue.length);
-    }
-
-    // Finally we will collect our user provided min/max values
-    // and the ones that we have calculated
-    var maxYValue = widget.maxYValue ?? calculatedMaxYValue;
-    var minYValue = calculatedMinYValue;
-
-    var unit = (widget.chartStyle ?? defaultCartesianChartStyle)
-        .gridStyle!
-        .yUnitValue!;
-    var maxYRange = maxYValue;
-    while (maxYRange % unit != 0) {
-      maxYRange++;
-    }
-
-    // We need to check for negative y values
-    var minYRange = minYValue;
-    if (minYValue.isNegative) {
-      while (minYRange % unit != 0) {
-        minYRange--;
-      }
-    } else {
-      // No negative y values, so min will be zero
-      minYRange = 0.0;
-    }
-
     // Now we can provide the chart details to the observer
     _controller = CartesianController(
-      data: [widget.data],
-      animateOnLoad: widget.animateOnLoad,
-      autoAnimate: widget.autoAnimate,
-      animation: AnimationController(
-        duration: const Duration(seconds: 1),
-        vsync: this,
-      ),
-      minValue: minYValue,
-      maxValue: maxYValue,
-      maxXRange: calculatedMaxXValue,
-      minXRange: calculatedMinXValue,
-      maxYRange: maxYRange,
-      minYRange: minYRange,
-    );
+        data: [widget.data],
+        animateOnLoad: widget.animateOnLoad,
+        autoAnimate: widget.autoAnimate,
+        animation: AnimationController(
+          duration: const Duration(seconds: 1),
+          vsync: this,
+        ),
+        rangeConstraints: (c) {
+          // For Bar charts, we normally don't consider x values, be it +ve or -ve
+          c.maxXValue = widget.data.barData.length.toDouble();
+          // Finally we will collect our user provided min/max values
+          // and the ones that we have calculated
+          c.maxXRange = c.maxXValue;
+          c.minXRange = c.minXValue;
+          var minYValue = c.minYValue;
+
+          var yUnit = (widget.chartStyle ?? defaultCartesianChartStyle)
+              .gridStyle!
+              .yUnitValue!;
```

Comment:
Are we sure it's always non null here?
I think even if `widget.chartStyle` is not null, the `gridStyle` inside ti might still be null.

Reply:
Good catch. 👍  Fixed the check